scrapely/extractors.py

```diff
@@ -303,13 +302,32 @@ def extract_price(txt):
     """
     txt = _NUMERIC_ENTITIES.sub(lambda m: unichr(int(m.groups()[0])), txt)
     m = _DECIMAL_RE.search(txt)
+    txt = txt.replace(' ', '')
+
     if m:
         value = m.group(1)
-        parts = _VALPARTS_RE.findall(value)
-        decimalpart = parts.pop(-1)
-        if decimalpart[0] == "," and len(decimalpart) <= 3:
-            decimalpart = decimalpart.replace(",", ".")
-        value = "".join(parts + [decimalpart]).replace(",", "")
+        decimalSeparator = 'point'  # defaults to point
```

Comment:
This whole section is not very pythonic. You should use `x in y` instead of `y.__contains__(x)`. You're using camelcase instead of snake case.

This implements the same logic but it should be faster (avoids unnecessary copying caused by slicing and does int comparisons instead of string comparisons where needed):
```python
    POINT, COMMA = 0, 1
    decimal_separator = POINT
    last_point_idx = value.rfind('.')
    last_comma_idx = value.rfind(',')
    if last_point_idx > 0 and last_comma_idx > 0:
        # If a number has both separators take the last one
        if last_comma_idx > last_point_idx:
            decimal_separator = COMMA
    elif last_comma_idx > 0:
        # If a number has only commas check the last one 
        first_comma_idx = value.find(',')
        if (first_comma_idx == last_comma_idx and
                len(value) - last_comma_idx != 4):
            decimal_separator = COMMA

    if decimal_separator == POINT:
        return value.replace(',', '')
    return value.replace('.', '').replace(',', '.')
```

Please be sure to add tests for this PR too